multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapListCache.java

```diff
@@ -111,15 +113,24 @@ public CompletionStage<Long> size(K key) {
       return cache.getAsync(key).thenApply(b -> b == null ? 0 : (long) b.size());
    }
 
-   public CompletionStage<Collection<V>> subList(int from, int to) {
-      throw new UnsupportedOperationException();
+   /**
+    * Returns the element at the given index. Index is zero-based.
+    * 0 means fist element. Negative index counts index from the tail. For example -1 is the last element.
+    *
+    * @param key, the name of the list
+    * @param index, the position of the element.
+    * @return Optional of the value. Empty if the key does not exist or the index is out of bounds.
+    */
+   public CompletionStage<Optional<V>> index(K key, long index) {
```

Comment:
Just nitpicking. We can always update this later, as it is only internal. But not sure if we should wrap it in an `Optional`, returning the value or null directly feels more in line with the queue/list operations.

Reply:
I was myself thinking about this too, concerning the public API. I can return null for sure